kolena/workflow/_datatypes.py

```diff
@@ -138,11 +139,12 @@ def serialize_value(value: Any) -> Any:
             raise ValueError(f"unsupported value type: '{type(value).__name__}' (value: {value})")
 
         items = [(field.name, getattr(self, field.name)) for field in dataclasses.fields(type(self))]
-        field_names = {field.name for field in dataclasses.fields(type(self))}
+        field_names = [field.name for field in dataclasses.fields(type(self))]
```

Comment:
Changing this to a list slows down the `key not in field_names` check a few lines after. Is there an ordered collection with O(1) lookup that we can use here?